packages/yarnpkg-nm/sources/hoist.ts

```diff
@@ -449,12 +452,19 @@ const getNodeHoistInfo = (rootNode: HoisterWorkTree, rootNodePathLocators: Set<L
     reason = `- self-reference`;
 
   if (isHoistable) {
-    isHoistable = !node.isWorkspace;
+    isHoistable = node.dependencyKind !== HoisterDependencyKind.WORKSPACE;
     if (outputReason && !isHoistable) {
       reason = `- workspace`;
     }
   }
 
+  if (isHoistable) {
+    isHoistable = node.dependencyKind !== HoisterDependencyKind.EXTERNAL_SOFT_LINK || node.dependencies.size === 0;
+    if (outputReason && !isHoistable) {
+      reason = `- portal with unhoisted dependencies`;
```

Comment:
What do you think about renaming it in the reason too?

```suggestion
      reason = `- external soft link with unhoisted dependencies`;
```

Reply:
Sure, thank you!